CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ project (pistache
 
 include(GNUInstallDirs)
 
-add_compile_options(-Wall -Wconversion -pedantic -Wextra -Wno-missing-field-initializers)
+add_compile_options(-Wall -Wextra -Wpedantic -Wconversion -Wno-sign-conversion -Wno-missing-field-initializers)
```

Comment:
Disabling sign-conversion could be dangerous. Do you have a good reason for this, other than to reduce build log verbosity?

Reply:
Yes. Have you ever tried compiling with Clang? Pistache does a lot of sign conversions (like `long - size_t`), and Clang throws hundreds of warnings, making it impossible to spot the important ones. 

Maybe it would be better to disable that warning only on Clang.

Reply:
Reading the [GCC docs](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html) it turns out that _Warnings about conversions between signed and unsigned integers are disabled by default in C++ unless -Wsign-conversion is explicitly enabled_, so this changes just makes sure that both GCC and Clang follow the same behavior. 

As for Meson, I have applied the same option in the [C++17 PR](https://github.com/pistacheio/pistache/pull/897/files?w=1#diff-30d8f6be6320feeacf686be94f48c70869b52630e01ea625f0f15adc0d57c3e4R20).

Reply:
I think it would probably be better if we don't suppress potentially useful warnings and instead and take care of the type conversions with `static_cast`, more appropriate types, or whatever the case may be.

Reply:
With the Autotools, and there may be some way of doing this with Meson, but the user can always override the compiler flags during configure time by setting CXXFLAGS which would be more appropriate for your situation where you are drowning in potentially innocuous warnings.

Reply:
I don't think that placing hundreds of explicit casts would be better than allowing implicit casts. Also, if GCC permits it there are a lot of good reasons (as they are god-level programmers). 

Unfortunately I don't have the time to investigate if any of the sign conversion warnings are actually useful, but since most of them look like "`implicit conversion changes signedness: 'ssize_t' (aka 'long') to 'size_t'`" I doubt that